react-native/index.d.ts

```diff
@@ -6132,51 +6036,51 @@ declare module "react" {
         /**
          * Sets value for key and calls callback on completion, along with an Error if there is any
          */
-        setItem(key: string, value: string, callback?: (error?: Error) => void): Promise<string>
+        setItem(key: string, value: string, callback?: (error?: Error) => void): Promise<void>
 
-        removeItem(key: string, callback?: (error?: Error) => void): Promise<string>
+        removeItem(key: string, callback?: (error?: Error) => void): Promise<void>
 
         /**
          * Merges existing value with input value, assuming they are stringified json. Returns a Promise object.
          * Not supported by all native implementation
          */
-        mergeItem(key: string, value: string, callback?: (error?: Error) => void): Promise<string>
+        mergeItem(key: string, value: string, callback?: (error?: Error) => void): Promise<void>
 
         /**
          * Erases all AsyncStorage for all clients, libraries, etc. You probably don't want to call this.
          * Use removeItem or multiRemove to clear only your own keys instead.
          */
-        clear(callback?: (error?: Error) => void): Promise<string>
+        clear(callback?: (error?: Error) => void): Promise<void>
 
         /**
          * Gets all keys known to the app, for all callers, libraries, etc
          */
-        getAllKeys(callback?: (error?: Error, keys?: string[]) => void): Promise<string>
+        getAllKeys(callback?: (error?: Error, keys?: string[]) => void): Promise<string[]>
 
         /**
          * multiGet invokes callback with an array of key-value pair arrays that matches the input format of multiSet
          */
-        multiGet(keys: string[], callback?: (errors?: Error[], result?: string[][]) => void): Promise<string>
+        multiGet(keys: string[], callback?: (errors?: Error[], result?: string[][]) => void): Promise<void>
```

Comment:
seems like this is what the code is doing: https://github.com/facebook/react-native/blob/eba356fb135171b49c2062e9ccac73335e251491/Libraries/Storage/AsyncStorage.js#L340